### PR TITLE
feat(aws-s3-stepfunctions): added logS3AccessLogs and S3BucketInterface

### PR DESCRIPTION
**Pull request by mickychetta**
*Issue #485 , if available:*

*Description of changes:*
-Added logS3AccessLogs construct prop and s3BucketInterface
-Updated README with new props
-Updated integ tests with no logging bucket
-Improve tests for 100% coverage, deleted repetitive conditions that were covered in input-validation.ts

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f7a4e8eacfdaf1172adf7a480b4477fc692a4660
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9a9a387b-209e-466c-88c0-9dd09e1e2663%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: f7a4e8eacfdaf1172adf7a480b4477fc692a4660
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c681e1f0-056b-43e1-990e-946832372f4d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: b439d44141b0632490ee6be29acd9e9a44da3438
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=af71ab4c-3997-4c35-926c-162a56f934e0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: b439d44141b0632490ee6be29acd9e9a44da3438
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=dccccc02-f00d-4250-ad32-bc92c72fdfca%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/test/integ.s3-stepfunctions-no-argument.ts [hunk 1]
```diff
@@ -32,7 +34,16 @@ const props: S3ToStepfunctionsProps = {
   logGroupProps: {
     removalPolicy: RemovalPolicy.DESTROY
   },
+  logS3AccessLogs: false,
+  deployCloudTrail: false
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/test/integ.s3-stepfunctions-no-argument.ts:38]: If we left this set to true (or default) on one test, I'm guessing it leaves behind a bucket?

**Review by biffgaut — COMMENTED** [review]
One question

File: source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/test/integ.s3-stepfunctions-no-argument.ts (hunk 1 above)
Comment (mickychetta) [reply source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/test/integ.s3-stepfunctions-no-argument.ts:38]: Yes, that is correct

Comment (biffgaut) [reply source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/test/integ.s3-stepfunctions-no-argument.ts:38]: Do we have a unit test that sets deployCloudTrail to true so we test it somewhere?

Comment (mickychetta) [reply source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/test/integ.s3-stepfunctions-no-argument.ts:38]: No, I turned it off so it doesn't leave any CloudTrail logging buckets for all tests. Did you want me to leave it on for a test?

**Review by biffgaut — APPROVED** [review]

**Pull request merged**